### PR TITLE
Adds GitHub Action to run unit tests

### PR DESCRIPTION
**Pull request by tomaroberts**
Adds `build_and_test.yml` which will replace existing `build.yml`
- builds dcm2niix
- sets up Python and dependencies
- runs pytest on `tests/` dir

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -23,6 +24,21 @@ jobs:
     steps:
     - uses: actions/checkout@v3
 
+    - name: Build dcm2niix
```
Comment (laurencejackson) [inline .github/workflows/build_and_test.yml:27]: Could this be replaced with a pip install of the pip package version? Or is there a particular reason you want to build form source? if you build from source in the MAP then do that here, but maybe an opportunity to simplify the build if you switch both to pip

**Review by laurencejackson — APPROVED** [review]
Approved. One comment that's optional.

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (tomaroberts) [reply .github/workflows/build_and_test.yml:27]: The pip version of dcm2niix is more than a year behind the source version. Previously when I was using an out-of-date dcm2niix I did actually run into some issues, so figure keeping it up-to-date is best.

**Review by laurencejackson — APPROVED** [review]
Re-approving as base changed

**Pull request merged**